lib/mayaUsd/utils/util.cpp

```diff
@@ -1953,35 +1978,143 @@ UsdMayaUtil::GetDictionaryFromArgDatabase(
     return args;
 }
 
-VtValue
-UsdMayaUtil::ParseArgumentValue(
-        const std::string& key,
-        const std::string& value,
-        const VtDictionary& guideDict)
+namespace {
+VtValue _ParseArgumentValue(const JsValue& jsValue, const VtValue& guideValue)
+{
+    if (guideValue.IsHolding<bool>()) {
+        if (jsValue.GetType() == JsValue::StringType) {
+            return VtValue(TfUnstringify<bool>(jsValue.GetString()));
+        } else {
+            return VtValue();
+        }
+    } else if (guideValue.IsHolding<std::string>()) {
+        if (jsValue.GetType() == JsValue::StringType) {
+            return VtValue(jsValue.GetString());
+        } else {
+            return VtValue();
+        }
+    } else if (guideValue.IsHolding<std::vector<VtValue>>()) {
+        if (jsValue.GetType() == JsValue::ArrayType) {
+            const JsArray&       jsArray = jsValue.GetJsArray();
+            std::vector<VtValue> guideVector = guideValue.Get<std::vector<VtValue>>();
+            std::vector<VtValue> vtValue;
+            if (guideVector.empty() && !jsArray.empty()) {
+                TF_CODING_ERROR("Guide vector is empty. Can not recursively parse.");
+                return VtValue(vtValue);
+            }
+            for (const JsValue& jsItem : jsArray) {
+                vtValue.push_back(_ParseArgumentValue(jsItem, guideVector.front()));
+            }
+            return VtValue(vtValue);
+        } else {
+            return VtValue();
+        }
+    }
+    return VtValue();
+}
+
+VtValue _ParseArgumentValue(const std::string& value, const VtValue& guideValue)
+{
+    // The export UI only has boolean and string parameters.
+    if (guideValue.IsHolding<bool>()) {
+        return VtValue(TfUnstringify<bool>(value));
+    } else if (guideValue.IsHolding<std::string>()) {
+        return VtValue(value);
+    } else if (guideValue.IsHolding<std::vector<VtValue>>()) {
+        // To prevent quoting issues in MEL scripts, we expect a string that is
+        // completely unquoted, containing an array of booleans, tokens, or
+        // other arrays:
+        //       [0,1,true,false]
+        //       [none,default]
+        //       [[useRegistry,UsdPreviewSurface],[displayColors,default]]
+        // So, to be able to parse this as valid JSON, we need to add quotes at
+        // each "[[," to alphanumeric transition. 
+        // NOTE: The array contains *tokens*, not freeform strings. So we can
+        //       skip spaces and consider all commas to be element separators.
```

Comment:
This might be too fragile. One option would be to make the options string true JSON, but since it is parsed by MEL on the Maya side, this will mean escaping a lot of double quotes.